### PR TITLE
Yarn2

### PR DESCRIPTION
**Pull request by PeteCoward**
just thought I'd pend a few minutes seeing if we could go to yarn 2

### CODE REVIEW

**Comment by PeteCoward** [comment]
Oooh we could webpack 5!

**Comment by PeteCoward** [comment]
oh dear, it breaks the override mechanism

File: webpack.config.js [hunk 1]
```diff
@@ -55,13 +57,21 @@ module.exports = (env) => {
...
                 Actions: path.resolve(__dirname, "src/js/actions"),
             },
+            plugins: [
+                PnpWebpackPlugin,
+            ],
+        },
+        resolveLoader: {
+            plugins: [
+                PnpWebpackPlugin.moduleLoader(module),
+            ],
```
Comment (PeteCoward) [inline webpack.config.js:74]: These change required by moving to yarn2 see [here](https://yarnpkg.com/advanced/migration#enable-the-pnp-plugin-when-using-webpack-4)

File: webpack.config.js [hunk 2]
```diff
@@ -11,9 +11,11 @@ const WebpackPwaManifest = require("webpack-pwa-manifest");
 const MiniCssExtractPlugin = require("mini-css-extract-plugin");
 const OptimizeCSSAssetsPlugin = require("optimize-css-assets-webpack-plugin");
 const TerserPlugin = require("terser-webpack-plugin");
+const PnpWebpackPlugin = require(`pnp-webpack-plugin`);
 
 const defaultEnvironmentConfiguration = require("./canoe-environment-default.js");
 const defaultProjectConfiguration = require("./canoe-project-default.js");
+process.traceDeprecation = true;
```
Comment (PeteCoward) [inline webpack.config.js:18]:
```suggestion

// uncomment this to find the source of webpack deprecation warnings
// process.traceDeprecation = true;
```

File: webpack.config.js [hunk 3]
```diff
@@ -183,8 +195,10 @@ module.exports = (env) => {
 
     const productionWebpackConfig = env && env.PRODUCTION ? require("./webpack.prod.js") : {};
 
-    const config = merge.strategy({
-        "resolve.modules": "prepend",
+    const config = mergeWithCustomize({
+        customizeArray: customizeArray({
+            'resolve.modules': 'prepend'
+          })
```
Comment (PeteCoward) [inline webpack.config.js:201]: webpack merge version upgrade from 4 to 5 requires this

**Comment by PeteCoward** [comment]
Please have a look at this @katevoss @sandroroux, once this release is out I'll put this up on staging.
We should try to end this project as up to date on dependencies as we can

**Comment by sandroroux** [comment]
@PeteCoward Looks cool. When running this branch, `./run-canoe.sh` warns:
```
(node:69183) [MODULE_NOT_FOUND] Error: html-webpack-inline-svg-plugin tried to access html-webpack-plugin, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
(node:69183) [MODULE_NOT_FOUND] Error: webpack-pwa-manifest tried to access html-webpack-plugin, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
```

The JavaScript console also throws:
image

**Comment by PeteCoward** [comment]
> @PeteCoward Looks cool. When running this branch, `./run-canoe.sh` warns:
> 
> ```
> (node:69183) [MODULE_NOT_FOUND] Error: html-webpack-inline-svg-plugin tried to access html-webpack-plugin, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
> (node:69183) [MODULE_NOT_FOUND] Error: webpack-pwa-manifest tried to access html-webpack-plugin, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
> ```
> 
> The JavaScript console also throws:
> image

Thanks, the MODUKE_NOT_FOUND warnings are know and I think there is a way for us to suppress them ( but I've lost the link )

The js error I will investigate!

**Comment by PeteCoward** [comment]
Now updated to RC-1.3, build warnings suppressed in [recommended](https://yarnpkg.com/configuration/yarnrc#packageExtensions) fashion, and gettext use [corrected](https://github.com/guillaumepotier/gettext.js/issues/37#issuecomment-636348035)

**Comment by PeteCoward** [comment]
please have a look @sandroroux and @katevoss would be good to get this in

**Review by katevoss — APPROVED** [review]
Looks good, no more JS errors

**Pull request merged**